### PR TITLE
Implemented the Remove Bars Feature

### PR DESCRIPTION
**Pull request by M4RKERSM4SK**
Discussion from the Community: The idea behind removing bars was to open up mines from the front without needing to go through frigate in beginner logic. There was also an argument that it gives ice more utility since reverse frigate can be a thing before wave beam.

This feature will be disabled by default

Details of Changes

- Added a boolean to the prime configuration
- Added a checkbox to the UI that toggles that configuration value
- Wrote a preset migration to write the default value for older presets migrating
- Edited the patch data factory to tell the patcher to remove specific objects in Great Tree Hall.
- Added Misc Resource to the Logic DB for this new feature
- Edited the correct node in Great Tree Hall to be able to use the Misc Resource
- Edited the bootstrap for the feature toggle

Tests

- All objects dealing with bars seems to be removed

```
0x002401B5 - bar
0x002401C0 - spinner auto-enable timer
0x00240216 - scan front
0x00240215 - scan back
```

- Boost Ball slot behaves as intended (Just pushes you out, is scan-able, seemingly has no issues)
- Music in room and transitioning to other surrounding rooms music seems fine
- Transitioned between all surrounding rooms and elevators.
- Enemies still chilling and spiking around, could kill them and the gravity sharks and the plants
- Water still good and slow in it without grav etc
- X ray platform still there
- Picked up an randomized item in Great Tree Chamber
- Spider Track still functional
- Switched between all visors and beams
- Did the stand-able trick up to Life Grove Tunnel

Maybe potential future cleanup needed

- One member of the community expressed the following during the implementation: Solved bars are by default disabled. Unsolved bars are on Gate Solved layer which gets removed if you disable the layer or remove the object. Might need to change 0x2401f6 for solved gate to re enable if any weirdness\issues are discovered.

### CODE REVIEW

**Comment by JeffGainsNGames** [comment]
Can you supply an image of the GUI updates

File: randovania/games/prime1/logic_database/Tallon Overworld.txt [hunk 1]
```diff
@@ -1465,7 +1465,7 @@ Extra - aabb: [-36.58473, 104.77827, -72.26208, 52.718994, 189.0148, 72.021576]
                   Morph Ball Bomb and Bomb Jump (Hypermode) and Complex Bomb Jump (Hypermode) and Combat/Scan Dash (Expert) and Movement (Hypermode) and Standable Terrain (Expert)
   > Next to Spinner
       Any of the following:
-          After Great Tree Hall Bars Unlocked
+          After Great Tree Hall Bars Unlocked or Enabled Remove Bars in Great Tree Hall
```
Comment (JeffGainsNGames) [inline randovania/games/prime1/logic_database/Tallon Overworld.txt:1468]: Please also include from opposite direction

File: randovania/games/prime1/gui/ui_files/preset_prime_qol.ui [hunk 2]
```diff
@@ -268,6 +268,20 @@
...
+            <widget class="QCheckBox" name="remove_bars_great_tree_hall_check">
+             <property name="text">
+              <string>Remove Bars in Great Tree Hall</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLabel" name="remove_bars_great_tree_hall_label">
+             <property name="text">
+              <string>Removes the boost ball bars obstacle in Tallon Overworld Great Tree Hall</string>
```
Comment (Miepee) [inline randovania/games/prime1/gui/ui_files/preset_prime_qol.ui:281]:
```suggestion
              <string>Removes the Boost Ball bars obstacle in Tallon Overworld's Great Tree Hall</string>
```

Potentially even with ", allowing free movement between the lower and upper levels of the room"

File: randovania/layout/preset_migration.py [hunk 3]
```diff
@@ -1216,6 +1216,12 @@ def _migrate_v91(preset: dict) -> dict:
     return preset
 
 
+def _migrate_v92(preset: dict) -> dict:
+    preset["configuration"]["remove_bars_great_tree_hall"] = False
```
Comment (Miepee) [inline randovania/layout/preset_migration.py:1220]: Gate this behind a prime 1 game check.

**Review by Miepee — CHANGES_REQUESTED** [review]
"One person from the Community" = UltiNaruto. No need to anonymize the name and it helps as a point of contact should that be needed for questions.

- Changelog entry missing

File: randovania/games/prime1/exporter/patch_data_factory.py [hunk 4]
```diff
@@ -733,6 +733,15 @@ def create_game_specific_data(self) -> dict:
                 }
             ]
 
+        # Remove Bars in Great Tree Hall
```
Comment (Miepee) [inline randovania/games/prime1/exporter/patch_data_factory.py:736]: Since you edited the pdf, you need to update all prime 1 exporting json files

**Comment by Miepee** [comment]
For anyone curious, I did a batch gen if this once and late magmoor/frigatte/upper mines did have meaningfully more progression.

**Comment by M4RKERSM4SK** [comment]
> Can you supply an image of the GUI updates

RemoveBarsGuiChange

File: randovania/games/prime1/gui/ui_files/preset_prime_qol.ui [hunk 5]
```diff
@@ -268,6 +268,23 @@
              </property>
             </widget>
            </item>
+           <item>
```
Comment (Miepee) [inline randovania/games/prime1/gui/ui_files/preset_prime_qol.ui:271]: after looking at this in the screnshot, can you move this to be  backwards labs and backwards upper mines?

Comment (M4RKERSM4SK) [reply randovania/games/prime1/gui/ui_files/preset_prime_qol.ui]: RemoveBarsGuiChange

**Pull request merged**